BaragonAgentService/src/test/java/com/hubspot/baragon/agent/PreferSameRackWeightingBalancedTest.java

```diff
@@ -0,0 +1,103 @@
+package com.hubspot.baragon.agent;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.jukito.JukitoRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+import com.hubspot.baragon.agent.config.BaragonAgentConfiguration;
+import com.hubspot.baragon.agent.handlebars.PreferSameRackWeightingHelper;
+import com.hubspot.baragon.models.BaragonAgentEc2Metadata;
+import com.hubspot.baragon.models.BaragonAgentMetadata;
+import com.hubspot.baragon.models.UpstreamInfo;
+
+
+@RunWith(JukitoRunner.class)
+public class PreferSameRackWeightingBalancedTest {
+
+  final List<String> availabilityZones = Arrays.asList("us-east-1a", "us-east-1a", "us-east-1b", "us-east-1c", "us-east-1e");
+
+  private Multiset<UpstreamInfo> generateUpstreams (List<String> availabilityZones) {
+    Multiset<UpstreamInfo> upstreams = HashMultiset.create();
```

Comment:
is there a reason this can't be a normal Set? I don't think we are using the Multiset functionality for this particular piece, and the code we are testing just expects any Collection